### PR TITLE
Upgrade Storybook and Webpack to fix vulnerable dependencies

### PR DESCRIPTION
**Pull request by michaelbjacobson**
This PR upgrades Storybook from v7 to v8, Webpack (also Webpack Dev Server) from v4 to v5, and Mock Service Worker from v0 to v2. These upgrades address vulnerable dependencies [ip](https://www.npmjs.com/package/ip) and [webpack-dev-server](https://www.npmjs.com/package/webpack-dev-server).

- https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#from-version-7x-to-800
- https://webpack.js.org/migrate/5/
- https://mswjs.io/docs/migrations/1.x-to-2.x/

### CODE REVIEW

File: client/components/mma/billing/Billing.stories.tsx [hunk 1]
```diff
@@ -35,22 +35,34 @@ export const NoSubscription: StoryObj<typeof Billing> = {
...
 	parameters: {
 		msw: [
-			rest.get('/api/me/mma', (_req, res, ctx) => {
-				return res(ctx.json(toMembersDataApiResponse()));
+			http.get('/api/me/mma', () => {
+				return new Response(JSON.stringify(toMembersDataApiResponse()), {
+					headers: {
+						'Content-Type': 'application/json',
+					},
+				})
```
Comment (rBangay) [inline client/components/mma/billing/Billing.stories.tsx:43]: could we use `HttpResponse` here as well like you have in contactUs.stories.tsx:

```
http.get('/api/me/mma', () => {
  return HttpResponse.json(toMembersDataApiResponse())
})
```

File: client/components/mma/billing/Billing.stories.tsx [hunk 2]
```diff
@@ -35,22 +35,34 @@ export const NoSubscription: StoryObj<typeof Billing> = {
...
-					ctx.json({
-						subscriptions: [],
-					}),
-				);
+			http.get('/mpapi/user/mobile-subscriptions', () => {
+				return new Response(JSON.stringify({subscriptions: []}), {
+					headers: {
+						'Content-Type': 'application/json',
+					},
+				})
```
Comment (rBangay) [inline client/components/mma/billing/Billing.stories.tsx:50]: same here re `HttpResponse`?

```
http.get('/mpapi/user/mobile-subscriptions', () => {
  return HttpResponse.json(toMembersDataApiResponse({subscriptions: []}))
})
```

File: client/components/mma/billing/Billing.stories.tsx [hunk 3]
```diff
@@ -35,22 +35,34 @@ export const NoSubscription: StoryObj<typeof Billing> = {
...
 			}),
-			rest.get('/api/invoices', (_req, res, ctx) => {
-				return res(ctx.json({ invoices: [] }));
+			http.get('/api/invoices', () => {
+				return new Response(JSON.stringify({invoices: []}), {
+					headers: {
+						'Content-Type': 'application/json',
+					},
+				})
```
Comment (rBangay) [inline client/components/mma/billing/Billing.stories.tsx:57]: another `HttpResponse` candidate?

```
http.get('/api/invoices', () => {
  return HttpResponse.json({invoices: []})
})
```

File: client/components/mma/billing/Billing.stories.tsx [hunk 4]
```diff
@@ -35,22 +35,34 @@ export const NoSubscription: StoryObj<typeof Billing> = {
...
 			}),
 
-			rest.get('/idapi/user', (_req, res, ctx) => {
-				return res(ctx.json(user));
+			http.get('/idapi/user', () => {
+				return new Response(JSON.stringify(user), {
+					headers: {
+						'Content-Type': 'application/json',
+					},
+				})
```
Comment (rBangay) [inline client/components/mma/billing/Billing.stories.tsx:65]: another `HttpResponse` candidate?

```
http.get('/idapi/user', () => {
  return HttpResponse.json(user)
})
```

File: client/components/mma/billing/Billing.stories.tsx [hunk 5]
```diff
@@ -65,32 +77,38 @@ export const WithSubscriptions: StoryObj<typeof Billing> = {
...
+				return new Response(JSON.stringify(toMembersDataApiResponse(
+					guardianWeeklyPaidByCard(),
+					digitalPackPaidByDirectDebit(),
+					newspaperVoucherPaidByPaypal(),
+					tierThree(),
+				)), {
+					headers: {
+						'Content-Type': 'application/json',
+					},
+				})
```
Comment (rBangay) [inline client/components/mma/billing/Billing.stories.tsx:90]: another `HttpResponse` candidate?

```
http.get('/api/me/mma', () => {
  return HttpResponse.json(toMembersDataApiResponse(
	  guardianWeeklyPaidByCard(),
	  digitalPackPaidByDirectDebit(),
	  newspaperVoucherPaidByPaypal(),
	  tierThree(),
  ))
})
```

File: client/components/mma/billing/Billing.stories.tsx [hunk 6]
```diff
@@ -65,32 +77,38 @@ export const WithSubscriptions: StoryObj<typeof Billing> = {
...
+						InAppPurchase,
+						InAppPurchaseIos,
+						InAppPurchaseAndroid,
+						PuzzleAppPurchaseAndroid,
+					],
+				}), {
+					headers: {
+						'Content-Type': 'application/json',
+					},
+				})
```
Comment (rBangay) [inline client/components/mma/billing/Billing.stories.tsx:104]: another HttpResponse candidate?

```
http.get('/mpapi/user/mobile-subscriptions', () => {
  return HttpResponse.json({
	  subscriptions: [
		  InAppPurchase,
		  InAppPurchaseIos,
		  InAppPurchaseAndroid,
		  PuzzleAppPurchaseAndroid,
	  ],
  })
})
```

File: client/components/mma/billing/Billing.stories.tsx [hunk 7]
```diff
@@ -65,32 +77,38 @@ export const WithSubscriptions: StoryObj<typeof Billing> = {
...
 			}),
-			rest.get('/api/invoices', (_req, res, ctx) => {
-				return res(ctx.json({ invoices: [guardianWeeklyCardInvoice] }));
+			http.get('/api/invoices', () => {
+				return new Response(JSON.stringify({ invoices: [guardianWeeklyCardInvoice] }), {
+					headers: {
+						'Content-Type': 'application/json',
+					},
+				})
```
Comment (rBangay) [inline client/components/mma/billing/Billing.stories.tsx:111]: another `HttpResponse` candidate?

```
http.get('/api/invoices', () => {
  return HttpResponse.json({ invoices: [guardianWeeklyCardInvoice] })
})
```

File: client/components/mma/accountoverview/AccountOverview.stories.tsx [hunk 8]
```diff
@@ -1,7 +1,7 @@
 import type { Meta, StoryObj } from '@storybook/react';
-import { rest } from 'msw';
-import { ReactRouterDecorator } from '../../../../.storybook/ReactRouterDecorator';
-import { featureSwitches } from '../../../../shared/featureSwitches';
+import { http, HttpResponse } from 'msw';
+import { ReactRouterDecorator } from '@/.storybook/ReactRouterDecorator';
```
Comment (rBangay) [inline client/components/mma/accountoverview/AccountOverview.stories.tsx:3]: thanks for doing this ... we could do with tidying up the import statements throughout the codebase at some point

**Review by rBangay — APPROVED** [review]
Pheww thanks for doing all this! nice work.

I added the do not merge label on here as a reminder about the election code freeze, but feel free to merge when you see fit 👍

**Review by tjmw — APPROVED** [review]
Looks really good - great work!

Agree with the comments Richard made about `HttpResponse`.

Was it intentional that `.yarn/releases/yarn-1.22.22.cjs` was committed? It's a pretty big file! I'm surprised we're not git ignoring `.yarn`, maybe we should be?

File: client/components/mma/accountoverview/AccountOverview.stories.tsx (hunk 8 above)
Comment (michaelbjacobson) [reply client/components/mma/accountoverview/AccountOverview.stories.tsx:3]: Yeah, I'm afraid I didn't do it very systematically here. Probably best to have another PR where we do this?

**Comment by michaelbjacobson** [comment]
> Looks really good - great work!
> 
> Agree with the comments Richard made about `HttpResponse`.
> 
> Was it intentional that `.yarn/releases/yarn-1.22.22.cjs` was committed? It's a pretty big file! I'm surprised we're not git ignoring `.yarn`, maybe we should be?

It was intentional to upgrade the Yarn version for the project. But I'm not sure whether there's any reason to include the `.yarn` directory in version control. I'm happy to add it to gitignore if you think that's best?

**Comment by tjmw** [comment]
> But I'm not sure whether there's any reason to include the .yarn directory in version control. I'm happy to add it to gitignore if you think that's best?

Yeah I'm not 100% sure, I had a poke around a few Guardian projects and some do gitignore `.yarn` (example [dotcom-rendering](https://github.com/guardian/dotcom-rendering/blob/72d966dbe7c1a3a006c7331bb9f536e5da2bf38b/.gitignore#L14) but it doesn't seem to be universal. But I don't think I've seen large files like this included in the repo before. With this change there are actually now two large files in this directory which total > 11M in size which doesn't seem quite right! Are they the actual yarn implementation for a couple of different versions?

**Pull request merged**